subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/IsolatedAntBuilderMemoryLeakIntegrationTest.groovy

```diff
@@ -95,6 +98,21 @@ class IsolatedAntBuilderMemoryLeakIntegrationTest extends AbstractIntegrationSpe
         groovyVersion << groovyVersions() * 3
     }
 
+    @Issue('https://github.com/gradle/gradle/issues/22172')
+    void 'CodeNarc does not fail with PermGen space error with Groovy 4'() {
+        given:
+        assumeGroovy4()
+        withCodenarc(groovyVersion, '3.1.0-groovy-4.0')
```

Comment:
While we're messing with CodeNarc versions, it probably makes sense to update [CodeNarcCoverage](https://github.com/gradle/gradle/blob/master/subprojects/code-quality/src/testFixtures/groovy/org/gradle/quality/integtest/fixtures/CodeNarcCoverage.groovy) with groovy4 compatible versions. Not sure if you've done this already in your groovy 4 branch, and I'm not saying you need to do this in this PR or anything. Just want to make sure this is on your radar.